### ISSUE TITLE
Interpret . and x as dot and cross product with vectors

### ISSUE THREAD

**Issue by jrkimber** [numworks/epsilon#2150]
Currently, to find the dot or cross product of two vectors, you need to go into the toolbox and select dot(,) or cross(,). This is time-consuming and a bit fiddly and does not match the notation used on paper. Would it be possible for the calculator to interpret x (times) and . as the cross and dot products when used with vectors. The calculator currently returns undefined for these operations so it would not conflict with anything else.

The desired behaviour would be that the following would return the vector (cross) product of the vectors:
![screenshot (35)](https://github.com/numworks/epsilon/assets/136351212/431116c4-e1aa-4ae6-bcd1-1c0c540c5c12)

And this would return the scalar (dot) product:
![screenshot (36)](https://github.com/numworks/epsilon/assets/136351212/802fabb4-3b7f-436e-aff6-98663e4af304)

Thanks

**Comment by adri1numworks** [comment]
Hello,
Thank you for your input. I've created an issue in our internal repository. This will not be a priority for now but we will let you know when we implement it.

**Comment by fbeleznay** [comment]
I would like to add three comments to this.

- Accepting these would be in contradiction to the definition of matrix multiplication. Since the notation Numworks is using for vectors and matrices is the same, one does not know if the tables on the screenshots are matrices or vectors. Numworks gives undefined, because as matrix prodicts, these are undefined.
- In case a shortcut like this is implemented, please keep in mind that on IB exams matrix operations are allowed, but vector operations are not.

As for dot product, the current version does work in a limited way, just we need to enter the first vector as a row vector, not a column vector.  The result will be a 1x1 matrix, not a number, but it is at least a shortcut.
![Screen Shot 2024-02-20 at 05 21 55](https://github.com/numworks/epsilon/assets/111730066/63a0465d-97c6-4fc5-adcc-bd0d2a4be78e)

**Comment by jrkimber** [comment]
Hi,

Regarding the points made:

- I think the fact that such a matrix multiplication is undefined _enables_ X to be used for cross product. Since multiplying two column vectors is undefined as a matrix multiplication, it could only be interpreted as a cross product. It could even, to make matters clearer, rewrite it as cross[u,v] after EXE is pressed, in the same way that it already rewrites other things. 
-Presumably this can just be locked out by IB exam mode in the same way as it is already.

Thanks,
James

**Comment by jrkimber** [comment]
Indeed. That's why I think rewriting it as dot[u,v] and cross[u,v] after EXE is pressed would be a good idea to make it clearer. It's clear then that it hasn't multiplied them as matrices. 

I teach A-Level Maths and Further Maths in England and we do both vector and matrix operations so it would be good if students could use the standard . and X notation.
